File: 11-container-with-most-water/maxArea.c
#include <stdio.h>
#include <limits.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

int maxArea(int *height, int heightSize){
    int result = 0, i = 0, j = heightSize-1;
    while(i<j){
        int tmp = height[i]<height[j]? height[i]:height[j];
        int area = tmp*(j-i);
        if(area > result){
            result = area;
        }
        height[i] < height[j] ? i++ : j--;
    }
    return result;
}

int main() {
    int height[] = {1,2,4,3};
//    printf("%lu", sizeof(height)/sizeof(height[0]));
    printf("%d\n", maxArea(height, sizeof(height)/sizeof(height[0])));
    return 0;
}

File: 011-container-with-most-water/maxArea.js
/**
 * @param {number[]} height
 * @return {number}
 */

var maxArea = function (height) {
    var result = 0,
        length = height.length,
        i = 0,
        j = length - 1;
    while (i < j) {
        var area = (j - i) * Math.min(height[i], height[j]);
        if (area > result) {
            result = area;
        }
        height[i] < height[j] ? i++ : j--;
    }
    return result;
}
console.log(maxArea([1, 2, 4, 3]));
